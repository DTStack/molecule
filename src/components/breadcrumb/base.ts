import { prefixClaName, getBEMElement } from 'mo/common/className';

export const defaultBreadcrumbClassName = prefixClaName('breadcrumb');
export const breadcrumbItemClassName = getBEMElement(
    defaultBreadcrumbClassName,
    'item'
);
export const breadcrumbLabelClassName = getBEMElement(
    defaultBreadcrumbClassName,
    'label'
);
