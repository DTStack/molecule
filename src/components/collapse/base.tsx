import {
    prefixClaName,
    getBEMElement,
    getBEMModifier,
} from 'mo/common/className';

export const defaultCollapseClassName = prefixClaName('collapse');
export const collapseItemClassName = getBEMElement(
    defaultCollapseClassName,
    'item'
);

export const collapsePaneClassName = getBEMElement(
    defaultCollapseClassName,
    'pane'
);

export const collapsingClassName = getBEMModifier(
    collapsePaneClassName,
    'collapsing'
);

export const collapseActiveClassName = getBEMModifier(
    defaultCollapseClassName,
    'active'
);
export const collapseHeaderClassName = getBEMElement(
    defaultCollapseClassName,
    'header'
);
export const collapseTitleClassName = getBEMElement(
    collapseHeaderClassName,
    'title'
);
export const collapseContentClassName = getBEMElement(
    defaultCollapseClassName,
    'content'
);
export const collapseExtraClassName = getBEMElement(
    defaultCollapseClassName,
    'extra'
);
