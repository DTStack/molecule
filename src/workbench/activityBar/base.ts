import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';
import { ID_ACTIVITY_BAR } from 'mo/common/id';

export const defaultClassName = prefixClaName(ID_ACTIVITY_BAR);
export const containerClassName = getBEMElement(defaultClassName, 'container');
export const normalItemsClassName = getBEMElement(defaultClassName, 'normal');
export const globalItemsClassName = getBEMElement(defaultClassName, 'global');
export const itemClassName = getBEMElement(defaultClassName, 'item');
export const itemCheckedClassName = getBEMModifier(itemClassName, 'checked');
export const itemDisabledClassName = getBEMModifier(itemClassName, 'disabled');
export const labelClassName = getBEMElement(defaultClassName, 'label');
export const indicatorClassName = getBEMElement(defaultClassName, 'indicator');
