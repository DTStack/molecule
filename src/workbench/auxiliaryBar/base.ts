import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';

export const defaultClassName = prefixClaName('auxiliaryBar');
export const containerClassName = getBEMElement(defaultClassName, 'container');

export const tabsClassName = getBEMElement(defaultClassName, 'tabs');
export const tabClassName = getBEMElement(tabsClassName, 'tab');

export const tabActiveClassName = getBEMModifier(tabClassName, 'active');
