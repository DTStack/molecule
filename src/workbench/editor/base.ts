import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';

export const defaultEditorClassName = prefixClaName('editor');
export const groupClassName = getBEMElement(defaultEditorClassName, 'group');
export const groupContainerClassName = getBEMElement(
    defaultEditorClassName,
    'group-container'
);
export const groupHeaderClassName = getBEMElement(
    defaultEditorClassName,
    'group-header'
);
export const groupTabsClassName = getBEMElement(
    defaultEditorClassName,
    'group-tabs'
);
export const groupActionsClassName = getBEMElement(
    defaultEditorClassName,
    'group-actions'
);
export const groupActionsItemClassName = getBEMElement(
    defaultEditorClassName,
    'group-actions-item'
);

export const groupActionItemDisabledClassName = getBEMModifier(
    groupActionsItemClassName,
    'disabled'
);

export const groupBreadcrumbClassName = getBEMElement(
    defaultEditorClassName,
    'group-breadcrumb'
);
