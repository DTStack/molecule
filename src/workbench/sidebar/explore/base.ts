import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';
import {
    ID_ACTIVITY_BAR,
    ID_SIDE_BAR,
    ID_EXPLORER,
    ID_FOLDER_TREE,
    ID_EDITOR_TREE,
} from 'mo/common/id';

const defaultClassName = prefixClaName(ID_SIDE_BAR);
const defaultExplorerClassName = prefixClaName(ID_EXPLORER, defaultClassName);
const activityBarItemFloatClassName = getBEMModifier(
    getBEMElement(defaultExplorerClassName, ID_ACTIVITY_BAR),
    'float'
);

const folderTreeClassName = prefixClaName(ID_FOLDER_TREE);
const folderTreeInputClassName = getBEMModifier(folderTreeClassName, 'input');
const folderTreeEditClassName = getBEMModifier(folderTreeClassName, 'editable');

const editorTreeClassName = prefixClaName(ID_EDITOR_TREE);
const editorTreeItemClassName = getBEMElement(editorTreeClassName, 'item');
const editorTreeGroupClassName = getBEMElement(editorTreeClassName, 'group');
const editorTreeFileNameClassName = getBEMElement(
    editorTreeItemClassName,
    'fileName'
);
const editorTreeFilePathClassName = getBEMElement(
    editorTreeItemClassName,
    'filePath'
);
const editorTreeActiveItemClassName = getBEMModifier(
    editorTreeItemClassName,
    'active'
);
const editorTreeCloseIconClassName = getBEMElement(
    editorTreeClassName,
    'close'
);
const editorTreeFileIconClassName = getBEMElement(editorTreeClassName, 'file');

export {
    defaultExplorerClassName,
    activityBarItemFloatClassName,
    folderTreeClassName,
    folderTreeInputClassName,
    folderTreeEditClassName,
    editorTreeClassName,
    editorTreeItemClassName,
    editorTreeGroupClassName,
    editorTreeFileNameClassName,
    editorTreeFilePathClassName,
    editorTreeActiveItemClassName,
    editorTreeCloseIconClassName,
    editorTreeFileIconClassName,
};
