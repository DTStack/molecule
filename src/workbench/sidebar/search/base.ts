import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';

const emptyTextValueClassName = getBEMModifier(
    getBEMElement(prefixClaName('search'), 'treeNode'),
    'empty'
);

const matchSearchValueClassName = getBEMModifier(
    getBEMElement(prefixClaName('search'), 'treeNode'),
    'match'
);

const deleteSearchValueClassName = getBEMModifier(
    getBEMElement(prefixClaName('search'), 'treeNode'),
    'delete'
);

const replaceSearchValueClassName = getBEMModifier(
    getBEMElement(prefixClaName('search'), 'treeNode'),
    'replace'
);

const treeContentClassName = getBEMElement(prefixClaName('search'), 'tree');

export {
    matchSearchValueClassName,
    emptyTextValueClassName,
    deleteSearchValueClassName,
    replaceSearchValueClassName,
    treeContentClassName,
};
