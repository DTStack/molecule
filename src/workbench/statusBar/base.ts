import { getBEMElement, prefixClaName } from 'mo/common/className';
import { IStatusBarItem } from 'mo/model';

export const statusBarClassName = prefixClaName('statusBar');
export const leftItemsClassName = getBEMElement(
    statusBarClassName,
    'left-items'
);
export const rightItemsClassName = getBEMElement(
    statusBarClassName,
    'right-items'
);
export const itemClassName = getBEMElement(statusBarClassName, 'item');

export function sortByIndex(a: IStatusBarItem, b: IStatusBarItem) {
    return a.sortIndex !== undefined && b.sortIndex !== undefined
        ? a.sortIndex - b.sortIndex
        : 0;
}
