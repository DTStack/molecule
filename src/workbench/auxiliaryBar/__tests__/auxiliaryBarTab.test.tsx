import { fireEvent, render } from '@testing-library/react';
import { AuxiliaryModel } from 'mo/model';
import React from 'react';
import AuxiliaryBarTab from '../auxiliaryBarTab';
import { tabClassName } from '../base';

const props = new AuxiliaryModel();

describe('The Auxiliary Bar Tab Component', () => {
    test('Should match snapshot', () => {
        expect(
            render(<AuxiliaryBarTab {...props} />).asFragment()
        ).toMatchSnapshot();
    });

    test('Should match snapshot in tabs mode', () => {
        expect(
            render(<AuxiliaryBarTab {...props} mode="tabs" />).asFragment()
        ).toMatchSnapshot();
    });

    test('Should match snapshot with data', () => {
        expect(
            render(
                <AuxiliaryBarTab
                    {...props}
                    mode="tabs"
                    data={[{ key: 1, title: 1 }]}
                />
            ).asFragment()
        ).toMatchSnapshot();
    });

    test('Should match snapshot with active data', () => {
        expect(
            render(
                <AuxiliaryBarTab
                    {...props}
                    mode="tabs"
                    data={[{ key: 1, title: 1 }]}
                    current={1}
                />
            ).asFragment()
        ).toMatchSnapshot();
    });

    test('Should expect call onClick', () => {
        const clickFn = jest.fn();
        const { container } = render(
            <AuxiliaryBarTab
                {...props}
                mode="tabs"
                data={[{ key: 1, title: 1 }]}
                onClick={clickFn}
            />
        );

        const dom = container.querySelector(`.${tabClassName}`);
        expect(dom).not.toBe(null);
        fireEvent.click(dom!);

        expect(clickFn).toBeCalled();
    });
});
