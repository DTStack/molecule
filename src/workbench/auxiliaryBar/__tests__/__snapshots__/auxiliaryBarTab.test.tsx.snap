// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Auxiliary Bar Tab Component Should match snapshot 1`] = `<DocumentFragment />`;

exports[`The Auxiliary Bar Tab Component Should match snapshot in tabs mode 1`] = `
<DocumentFragment>
  <ul
    class="mo-auxiliaryBar__tabs"
  />
</DocumentFragment>
`;

exports[`The Auxiliary Bar Tab Component Should match snapshot with active data 1`] = `
<DocumentFragment>
  <ul
    class="mo-auxiliaryBar__tabs"
  >
    <li
      class="mo-auxiliaryBar__tabs__tab mo-auxiliaryBar__tabs__tab--active"
    >
      1
    </li>
  </ul>
</DocumentFragment>
`;

exports[`The Auxiliary Bar Tab Component Should match snapshot with data 1`] = `
<DocumentFragment>
  <ul
    class="mo-auxiliaryBar__tabs"
  >
    <li
      class="mo-auxiliaryBar__tabs__tab"
    >
      1
    </li>
  </ul>
</DocumentFragment>
`;
